pkg/deployments/tasks/20221101-l2-gauge-checkpointer/test/task.fork.ts

```diff
@@ -0,0 +1,234 @@
+            address: gaugeInput[0],
+            weight: await gaugeController['gauge_relative_weight(address)'](gaugeInput[0]),
+            expectedCheckpoints: gaugeInput[1],
+          };
+        })
+      );
+    };
+    const polygonRootGaugesData: GaugeData[] = await getGaugesData(polygonRootGauges);
+    const arbitrumRootGaugesData: GaugeData[] = await getGaugesData(arbitrumRootGauges);
+
+    // There are no optimism gauges that are correctly added to the Gauge Adder and the Gauge Controller at this point.
+    gauges.set(GaugeType.Polygon, polygonRootGaugesData);
+    gauges.set(GaugeType.Arbitrum, arbitrumRootGaugesData);
+  });
+
+  before('add gauges to checkpointer', async () => {
+    await Promise.all(
+      Array.from(gauges).map(([gaugeType, gaugesData]) => {
+        L2GaugeCheckpointer.addGauges(
+          gaugeType,
+          gaugesData.map((gaugeData) => gaugeData.address)
+        );
+      })
+    );
+  });
+
+  before('grant checkpoint permission to fees gauge checkpointer', async () => {
+    // Any gauge works; we just need the interface.
+    const gauge = await task.instanceAt('IStakelessGauge', gauges.get(GaugeType.Polygon)![0].address);
+    const govMultisig = await impersonate(GOV_MULTISIG, fp(100));
+
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await authorizerAdaptor.getActionId(gauge.interface.getSighash('checkpoint')),
+        L2GaugeCheckpointer.address
+      );
+  });
+
+  it('checks that gauges were added correctly', async () => {
+    for (const [gaugeType, gaugesData] of gauges.entries()) {
+      expect(await L2GaugeCheckpointer.getTotalGauges(gaugeType)).to.be.eq(gaugesData.length);
+    }
+  });
+
+  describe('getTotalBridgeCost', () => {
+    context('when threshold is 1', () => {
+      itChecksTotalBridgeCost(fp(1));
+    });
+
+    context('when threshold is 0.0001', () => {
+      itChecksTotalBridgeCost(fp(0.0001));
+    });
+
+    context('when threshold is 0', () => {
+      itChecksTotalBridgeCost(fp(0));
+    });
+  });
+
+  describe('checkpoint', () => {
+    const checkpointedGauges: GaugeData[] = [];
+    let gaugeDataAboveMinWeight: GaugeData[] = [];
+    let minRelativeWeight: BigNumber;
+
+    // Gauges won't be checkpointed twice, so when the threshold is lowered and more gauges get above the threshold
+    // we need to filter out those that have already been checkpointed.
+    beforeEach('get non-checkpointed gauges above min weight', () => {
+      gaugeDataAboveMinWeight = [
+        ...getGaugeDataAboveMinWeight(GaugeType.Polygon, minRelativeWeight),
+        ...getGaugeDataAboveMinWeight(GaugeType.Arbitrum, minRelativeWeight),
+      ];
+    });
+
+    afterEach('mark checkpointed gauges to consider in the next iteration', () => {
+      checkpointedGauges.push(...gaugeDataAboveMinWeight);
+    });
+
+    context('when threshold is 1', () => {
+      minRelativeWeight = fp(1);
+      itCheckpointsGaugesAboveRelativeWeight();
```

Comment:
I'd instead check here that this returns 0 (since no gauge fulfills this criteria). It's good to have a couple tests with hardcoded values and not depend too much on the 'smart' tests.